Update flash message + parent card stylez

brendansudol opened the pull request:
**How**:
* mapped flash message keys to a (smaller) whitelist of categories
for styling
* removed spacing related properties in card and alert SCSS in favor
of utility spacing classes in the HTML to avoid redundancies

Preview:
<img width="655" alt="screen shot 2016-09-30 at 12 26 42 pm" src="https://cloud.githubusercontent.com/assets/1060893/18999135/0e74662a-870a-11e6-9f7b-aada454ae5a3.png">

jessieay commented:
Nice! are we using the web design standards in this proj? This style looks similar but slightly different from the uswds flashes: https://standards.usa.gov/alerts/

jessieay commented on app/assets/stylesheets/components/_alert.scss:
```diff
@@ -1,43 +1,20 @@
-  &::before {
-    content: url(image-path('alert/ico-check.svg'));
-  }
+  color: #088a22;
```
should we name these so they can more easily be re-used?

brendansudol replied:
i think we should name them if we end of using them in at least one other place; i prefer not to create variables for single usage css

jessieay commented on app/helpers/application_helper.rb:
```diff
@@ -16,6 +16,13 @@ def tooltip(text)
 
+  def alert_category(key)
+    key = key.to_s.downcase
+    return 'oops' if key.in? %w(error alert)
```
this is interesting / cool -- would it make sense to update the controllers to use one type of alert or the other rather than looking for two types and giving them the same style?

jessieay commented on app/helpers/application_helper.rb:
```diff
@@ -16,6 +16,13 @@ def tooltip(text)
+  def alert_category(key)
+    key = key.to_s.downcase
+    return 'oops' if key.in? %w(error alert)
+    return 'success' if key.in? %w(notice success)
```
same comment here -- I assume we own all the code that creates theses flash messages. would it make sense to just updated them to all be `notice` or `success` rather than forcing them to be the same via this helper?

monfresh replied:
+1 to @jessieay's comments. We favor the "Tell, Don't Ask" principle. This helper can be removed by standardizing the flash key names and creating matching CSS classes.

brendansudol replied:
i think that's fine -- i was worried that devise was auto-creating some flash messages with diff flash types that that may not be true; will explore!

brendansudol replied:
okiedokie

jessieay commented on app/views/layouts/card.html.slim:
```diff
@@ -1,6 +1,6 @@
 = content_for :content do
   .container.sm-pt4
-    div class="card #{yield(:card_cls)}"
+    div class="px2 py3 sm-py5 sm-px6 mx-auto mb5 border-box card #{yield(:card_cls)}"
```
wow this is a lot of classes! what do you call this style of css-ing?

brendansudol replied:
this is called atomic css! single purpose, utility classes -- the idea is that they should do one thing and do it well, should be simple and obvious to use, and should operate independently; background reading - http://mrmrs.io/writing/2016/03/24/scalable-css/ :)

brendansudol replied:
excited to introduce you to this system! it might be the best thing invented since emojis 😉 

additional articles if you're curious -- https://blog.colepeters.com/building-and-shipping-functional-css/ https://medium.com/buzzfeed-design/how-i-learned-to-stop-worrying-and-love-the-atomic-class-98d6ccc45781#.ktwfxs78l

jessieay replied:
very cool! excited to see this in action

brendansudol commented:
@jessieay - re: your question about the us web design standards, we are not explicitly using them. we are leveraging ideas and elements from it, like colors and fonts, but there's a lot in uswds that we don't need and don't want to unnecessarily include so that we can keep our pages nice and light and fast!

brendansudol commented:
addressed comments -- i didn't update flash message keys throughout because i think there is value to having more than a binary success / fail (i.e., sometimes you just want a FYI flash message), but we now handle appropriate styling and images without the need of a helper method, with good fallback stylez just in case :)
